src/stash/src/persist.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::{hash_map, HashMap};
+use std::fmt;
+use std::hash::Hash;
+use std::vec;
+
+use ore::metrics::MetricsRegistry;
+use persist::client::{RuntimeClient, StreamWriteHandle};
+use persist::indexed::Snapshot;
+use persist::runtime::{self, RuntimeConfig};
+use persist::storage::{Blob, Log};
+use persist_types::Codec;
+use timely::progress::Antichain;
+
+use crate::{Stash, StashError, StashOp};
+
+type Error = StashError<persist::error::Error>;
+
+/// A stash backed by a [`persist`] collection.
+#[derive(Debug)]
+pub struct PersistStash<K, V> {
+    _client: RuntimeClient,
+    write_handle: StreamWriteHandle<K, V>,
+    data: HashMap<K, V>,
+    timestamp: u64,
+}
+
+impl<K, V> PersistStash<K, V>
+where
+    K: Codec + Clone + Eq + Hash + Ord + fmt::Debug,
+    V: Codec + Clone + Eq + Hash + Ord + fmt::Debug,
+{
+    /// Opens the stash specified by the provided blob and log.
+    pub fn open<B, L>(
+        blob: B,
+        log: L,
+        metrics_registry: &MetricsRegistry,
+    ) -> Result<PersistStash<K, V>, Error>
+    where
+        B: Blob + Send + 'static,
+        L: Log + Send + 'static,
+    {
+        let client = runtime::start(
+            RuntimeConfig::default(),
+            log,
+            blob,
+            build_info::DUMMY_BUILD_INFO,
```

Comment:
this needs to be plumbed though the open call :). It's fine for the test code to pass build_info::DUMMY_BUILD_INFO in for it though